test/box-tap/gh-6647-yield-gc-hook.test.lua

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env tarantool
+
+local tap = require('tap')
+local test = tap.test('gh-6647-yield-gc-hook')
+local fio = require('fio')
+test:plan(1)
+
+local tarantool_bin = arg[-1]
+local PANIC = 256
+
+local function run_script(code)
+    local dir = fio.tempdir()
+    local script_path = fio.pathjoin(dir, 'script.lua')
+    local script = fio.open(script_path, {'O_CREAT', 'O_WRONLY', 'O_APPEND'},
+        tonumber('0777', 8))
+    script:write(code)
+    script:write("\nos.exit(0)")
+    script:close()
+    local cmd = [[/bin/sh -c 'cd "%s" && "%s" ./script.lua 2> /dev/null']]
+    local res = os.execute(string.format(cmd, dir, tarantool_bin))
+    fio.rmtree(dir)
+    return res
+end
+
+local space_trigger = [[
+box.cfg({listen = 3301})
```

Comment:
Our testing is parallel and there is no guarantee (or even a good probability) that any particular TCP port is free. There are two well-known ways to overcome it:

* Use a Unix domain socket.
* Bind on zero TCP port (look https://github.com/tarantool/tt/pull/112 for details).

I suggest the former, because it is simpler. It has its own limitation: 103 (Mac OS) or 107 (Linux) symbols in the socket path. However it unlikely will hit us here.